### PR TITLE
sample_data function and tests

### PR DESCRIPTION
**Pull request by moniquewong**
This pull request contains sample_data function and tests

Current tasks remaining: 
- [x] need to complete tests to maximize code coverage

### CODE REVIEW

File: R/sample_data.R [hunk 1]
```diff
@@ -1,3 +1,65 @@
...
+#' @return dataframe
+#' @export
+summarize_data <- function(df) {
+  # create data frame with column names and first record
+  result <- as.data.frame(t(dplyr::slice(df, 1)))
+  result['columns'] = colnames(df)
+  result <- dplyr::select(dplyr::mutate(result, sample_record = V1), columns, sample_record)
+
+  # add data types of columns
+  result['data_type'] = sapply(df, class)
```
Comment (mirohu) [inline R/sample_data.R]: For some reason, the .csv file it generates doesn't contain a column with data type. Not sure if my suggestion would solve it..
```suggestion
  result['data_type'] <- sapply(df, class)
```

File: R/sample_data.R [hunk 2]
```diff
@@ -1,3 +1,65 @@
...
+#'
+#' @param df the dataframe object to analyze
+#'
+#' @return dataframe
+#' @export
+summarize_data <- function(df) {
+  # create data frame with column names and first record
+  result <- as.data.frame(t(dplyr::slice(df, 1)))
+  result['columns'] = colnames(df)
+  result <- dplyr::select(dplyr::mutate(result, sample_record = V1), columns, sample_record)
```
Comment (mirohu) [inline R/sample_data.R]: The generated .csv file doesn't show the `colname` and `sample_record` at my end

File: R/sample_data.R [hunk 3]
```diff
@@ -1,3 +1,65 @@
...
+#' an example record, column data types and summary statistics for different data types
+#'
+#' @param df the dataframe object to analyze
+#'
+#' @return dataframe
+#' @export
+summarize_data <- function(df) {
+  # create data frame with column names and first record
+  result <- as.data.frame(t(dplyr::slice(df, 1)))
+  result['columns'] = colnames(df)
```
Comment (mirohu) [inline R/sample_data.R]:
```suggestion
  result['columns'] <- colnames(df)
```

**Review by mirohu — CHANGES_REQUESTED** [review]
It looks good! There might be some minor syntax error as the output table is missing something. Can you please double-check that?

File: R/sample_data.R (hunk 2 above)
Comment (moniquewong) [reply R/sample_data.R]: Updated tests to make this change

**Pull request merged**